pkg/components/gtime/gtime.go

```diff
@@ -32,5 +32,47 @@ func ParseInterval(interval string) (time.Duration, error) {
 		return now.Sub(now.AddDate(-num, 0, 0)), nil
 	}
 
-	return 0, fmt.Errorf("ParseInterval: invalid duration %q", interval)
+	return 0, fmt.Errorf("invalid interval %q", inp)
+}
+
+// ParseDuration parses a duration with support for all units that Grafana uses.
+func ParseDuration(inp string) (time.Duration, error) {
+	num, period, err := parse(inp)
+	if err != nil {
+		return 0, err
+	}
+	if period == "" {
+		return time.Duration(num), nil
+	}
+
+	// The average number of days in a year, using the Julian calendar
+	const daysInAYear = 365.25
```

Comment:
If we have time constants, we can then just multiply num with the relevant time constant and reduce the complexity of the maths.

Since `time.Duration` is nanosecond resolution, the loss of sub-nanoseconds when converting from float64 isn't too concerning even when multiplying with very big numbers 😉

```suggestion
	// The average number of days in a year, using the Julian calendar
	const daysInAYear = 365.25
	const day = 24 * time.Hour
	const week = 7 * day
	const year = time.Duration(float64(day) * daysInAYear)
	const month = year / 12
```